src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs

```diff
@@ -133,5 +133,34 @@ internal bool HasClientCredentials
         /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
         /// </summary>
         public string? UserAssignedManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow app-only tokens?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration you'll need to opt-in for the idtyp claim.
+        /// </remarks>
+        public bool ApiAllowsAppOnlyTokens { get; set; } = true;
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow guest accounts?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration for your web API, you'll need to opt-in for the acct claim.
+        /// </remarks>
+        public bool ApiAllowsGuestAccounts { get; set; } = true;
+
+        /// <summary>
+        /// Applies to web apps and web APIs. Enables to provide the list
+        /// of tenant Ids the user of whom are allowed to call the web API
+        /// or log-in to the web app.
+        /// </summary>
+        public IEnumerable<string>? AllowedTenantIds { get; set; }
+
+        /// <summary>
+        /// Applies to web APIs. Enables to provide the list
+        /// of client applications allowed to call this API.
+        /// </summary>
+        public IEnumerable<string>? AllowedClientApplications { get; set; }
```

Comment:
Since we recommend against using appids directly for authZ, I would not offer such method.

Reply:
This is an oid check

Reply:
@hpsin : should it be `AllowedClientOids`? And in that case would it be more for the web APIs called with a user access token?

Reply:
we should talk it over. Is oid check really needed?

Reply:
both app only tokens and user+app otkens have an oid.  It's not a particularly usable setup,. so consider removing it entirely.